src/modules/Dropdown/Dropdown.js

```diff
@@ -52,28 +52,17 @@ export default class Dropdown extends Component {
 
   static getAutoControlledStateFromProps(nextProps, computedState, prevState) {
     // These values are stored only for a comparison on next getAutoControlledStateFromProps()
-    const derivedState = { __value: nextProps.value, __options: nextProps.options }
+    const derivedState = { __options: nextProps.options, __value: computedState.value }
```

Comment:
This should use `computedState.value` as we want to compute new `selectedIndex` based the component's state